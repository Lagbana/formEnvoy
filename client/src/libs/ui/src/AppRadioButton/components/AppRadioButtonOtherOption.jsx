import * as React from 'react';
import { TouchableOpacity, StyleSheet, View, Text } from 'react-native';
import { Typography, Colors } from 'libs/ui/src/WebButton/node_modules/libs/ui/src/WebLoading/node_modules/colibri/styles';
import { AppAnimatedView } from '../../AppAnimatedView/AppAnimatedView';
import { AppRadioButtonDirection, AppRadioButtonOptionOtherType } from '../AppRadioButton.interface';
import { AppIcon } from '../../AppIcon';

export interface AppRadioButtonOtherOptionProps<T> {
	option: AppRadioButtonOptionOtherType<T>;
	selectedValue?: T | null;
	direction: AppRadioButtonDirection;
	onSelect?: (value: T) => void;
	animation: {
		opacity: number;
		translateY: number;
	};
}

export const AppRadioButtonOtherOption: <T = any>(props: AppRadioButtonOtherOptionProps<T>) => React.ReactElement = (props) => {

  const { 
		direction,
		selectedValue,
		onSelect,
		option: { value, icon, label }, 
		animation: { translateY, opacity } 
	} = props;

	const styles = StyleSheet.create({
		box: {
      width: direction === 'row' ? '50%' : '100%',
      padding: 7,
		},
		image: {
      width: '100%',
      aspectRatio: 1,
      borderBottomRightRadius: 14,
      borderBottomLeftRadius: 14,
		},
		default: {},
		selected: {},
		label: {
			...Typography.labelDefault,
			color: Colors.gray,
		},
		selectedLabel: {},
		icon: {
			color: Colors.green,
			marginRight: 10,
		},
		wrapper: {
			flexDirection: 'row',
			alignItems: 'center',
			paddingVertical: 20,
			paddingLeft: 10,
			paddingRight: 42
		}
	});

	const handleSelect = () => { if (onSelect) { onSelect(value); } }

	const isActive = selectedValue === value;

	const iconDisplay = (icon) ? <AppIcon
		name={icon || 'check'}
		size={22}
		style={[
			styles.icon,
			{ opacity: isActive ? 1 : 0 },
		]}
	/> : null;
	return (
		<AppAnimatedView
			style={[
				styles.box, 
				/*
				{ 
					opacity,
					translate: [
						{
							translateY
						}
					]
				 }
				 */
			]}
		>
			<TouchableOpacity
				style={[
					styles.default,
					isActive ? styles.selected : {},
				]}
				onPress={handleSelect}
				onLongPress={handleSelect}
			>
				<View
					style={styles.wrapper}
				>
					{iconDisplay}
					<Text
						style={[
							styles.label,
							isActive? styles.selectedLabel : {},
						]}
					>
						{label}
					</Text>
				</View>
			</TouchableOpacity>
		</AppAnimatedView>
	);
	
}