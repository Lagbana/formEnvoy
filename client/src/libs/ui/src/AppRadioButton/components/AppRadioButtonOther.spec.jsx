import React from 'react'
import { shallow, configure } from 'enzyme'
import toJson from 'enzyme-to-json'
import Adapter from 'enzyme-adapter-react-16'
import { AppRadioButtonOtherOption } from './AppRadioButtonOtherOption'

configure({ adapter: new Adapter() })

test('<AppRadioButtonOtherOption /> renders correctly', () => {
  const component = shallow(
    <AppRadioButtonOtherOption
      animation={{ opacity: 100, translateY: 0 }}
      option={{ value: 0, label: 'test label', other: true }}
      direction={'row'}
    />
  )
  expect(toJson(component)).toMatchSnapshot()
})
