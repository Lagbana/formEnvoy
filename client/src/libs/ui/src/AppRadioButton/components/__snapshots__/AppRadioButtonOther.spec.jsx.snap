// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppRadioButtonOtherOption /> renders correctly 1`] = `
<ForwardRef
  style={
    Array [
      Object {
        "padding": 7,
        "width": "50%",
      },
    ]
  }
>
  <ForwardRef
    onLongPress={[Function]}
    onPress={[Function]}
    style={
      Array [
        Object {},
        Object {},
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "paddingLeft": 10,
          "paddingRight": 42,
          "paddingVertical": 20,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#808898",
              "fontFamily": "ObjektivMk1-Bold",
              "fontSize": 18,
              "lineHeight": 19.8,
              "paddingTop": 4.5,
            },
            Object {},
          ]
        }
      >
        test label
      </Text>
    </View>
  </ForwardRef>
</ForwardRef>
`;
